cmd/soroban-rpc/lib/preflight/src/fees.rs

```diff
@@ -138,13 +141,14 @@ fn calculate_host_function_soroban_resources(
         })
         .sum();
 
-    let contract_events_size_bytes = calculate_event_size_bytes(events)?;
+    let contract_events_size_bytes =
+        calculate_event_size_bytes(events).context("cannot calculate events size")?;
 
     // Add a 15% leeway with a minimum of 50k instructions
-    let instructions = max(
-        budget.get_cpu_insns_consumed()? + 50000,
-        budget.get_cpu_insns_consumed()? * 115 / 100,
-    );
+    let budget_instructions = budget
+        .get_cpu_insns_consumed()
+        .context("cannot get instructions consumed")?;
```

Comment:
```suggestion
        .context("cannot calculate instructions consumed")?;
```
Nit: Might be more accurate? Because, "cannot get" could mean the actual instructions are not available to be consumed. Probably fine either way tho.

Reply:
It actually gets them, it only fails if it cannot borrow.

Reply:
Sure, that's fine. I mean that this text we're returning to the user could be misinterpreted.

Reply:
In reality it's something which should be considered an internal error though.

Reply:
ok then